disnake/embeds.py

```diff
@@ -302,6 +316,27 @@ def __bool__(self) -> bool:
             )
         )
 
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, Embed) and all(
+            (
+                (self.title or None) == (other.title or None),
+                (self.description or None) == (other.description or None),
+                (self.url or None) == (other.url or None),
+                (self._timestamp or None) == (other._timestamp or None),
+                (self._colour or None) == (other._colour or None),
+                (self._thumbnail or None) == (other._thumbnail or None),
+                (self._video or None) == (other._video or None),
+                (self._provider or None) == (other._provider or None),
+                (self._author or None) == (other._author or None),
+                (self._image or None) == (other._image or None),
+                (self._footer or None) == (other._footer or None),
+                (self._fields or None) == (other._fields or None),
```

Comment:
These should be restructured as `a == b and c == d and e == f ...` to make use of short-circuit evaluation

Reply:
This could possibly even be refactored to loop over `__slots__`:
`all(getattr(self, slot) == getattr(other, slot) for slot in slots)`. If I'm not mistaken, that should innately short-circuit, too.

Reply:
Yep, you're right
```py
>>> def generate():
...     for x in [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]:
...         print(x)
...         yield x
...
>>> all(y < 5 for y in generate())
1
2
3
4
5
False
>>>
```